go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -311,6 +361,287 @@ func registerUser(userAddress string, event []UserEventType, subsIdStr string) {
+			if skipThisSubscription {
+				continue
+			}
+			if len(returnAddr) > 0 {
+				subsIdStr := strconv.Itoa(subsId)
+
+				var distanceNotif SubscriptionNotification
+				distanceNotif.DistanceCriteria = distanceCheck.Subscription.Criteria
+				distanceNotif.IsFinalNotification = false
+				distanceNotif.Link = distanceCheck.Subscription.Link
+				var terminalLocationList []TerminalLocation
+				for terminalAddr, distanceInfo := range returnAddr {
+					var terminalLocation TerminalLocation
+					terminalLocation.Address = terminalAddr
+					var locationInfo LocationInfo
+					locationInfo.Latitude = nil
+					locationInfo.Latitude = append(locationInfo.Latitude, distanceInfo.Latitude)
+					locationInfo.Longitude = nil
+					locationInfo.Longitude = append(locationInfo.Longitude, distanceInfo.Longitude)
+					locationInfo.Shape = 2
+					seconds := time.Now().Unix()
+					var timestamp TimeStamp
+					timestamp.Seconds = int32(seconds)
+					locationInfo.Timestamp = &timestamp
+					terminalLocation.CurrentLocation = &locationInfo
+					retrievalStatus := RETRIEVED
+					terminalLocation.LocationRetrievalStatus = &retrievalStatus
+					terminalLocationList = append(terminalLocationList, terminalLocation)
+				}
+				distanceNotif.TerminalLocation = terminalLocationList
+				distanceNotif.CallbackData = distanceCheck.Subscription.ClientCorrelator
+				var inlineDistanceSubscriptionNotification InlineSubscriptionNotification
+				inlineDistanceSubscriptionNotification.SubscriptionNotification = &distanceNotif
+				sendSubscriptionNotification(distanceCheck.Subscription.CallbackReference.NotifyURL, inlineDistanceSubscriptionNotification)
+				log.Info("Distance Notification"+"("+subsIdStr+") For ", returnAddr)
+			}
+		}
+	}
+}
+
+func checkNotificationAreaCircle(addressToCheck string) {
+
+	//only check if there is at least one subscription
+	mutex.Lock()
+	defer mutex.Unlock()
+	//check all that applies
+	for subsId, areaCircleCheck := range areaCircleSubscriptionMap {
+		if areaCircleCheck != nil && areaCircleCheck.Subscription != nil {
+			//decrement the next time to send a message
+			areaCircleCheck.NextTts--
+			if areaCircleCheck.NextTts > 0 {
+				continue
+			} else { //restart the nextTts and continue processing to send notification or not
+				areaCircleCheck.NextTts = areaCircleCheck.Subscription.Frequency
+			}
+
+			//loop through every reference address
+			for _, addr := range areaCircleCheck.Subscription.Address {
+				if addr != addressToCheck {
+					continue
+				}
+				//check if address is already inside the area or not based on the subscription
+				var withinRangeParam gisClient.TargetRange
+				withinRangeParam.Latitude = areaCircleCheck.Subscription.Latitude
+				withinRangeParam.Longitude = areaCircleCheck.Subscription.Longitude
+				withinRangeParam.Radius = areaCircleCheck.Subscription.Radius
+
+				withinRangeResp, _, err := gisAppClient.GeospatialDataApi.GetWithinRangeByName(context.TODO(), addr, withinRangeParam)
```

Comment:
Should we be blocking the entire location service while waiting for this response?

Reply:
same comment as above regarding adding go routines... same conclusion should apply

Reply:
I'm okay with leaving it as is. If we see performance degradations or longer processing delays then we can address.